### PR TITLE
Functional tests - Add new test 'FO > Check slider'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | Add new test 'FO > Check slider'
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| Related PRs       | no
| How to test?      | no need
| Possible impacts? | no

### CODE REVIEW

File: tests/UI/campaigns/functional/FO/06_homePage/01_checkSlider.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  after(async () => {
+    await helper.closeBrowserContext(browserContext);
+  });
+
+  it('should open the shop page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'openShopFO', baseContext);
+
+    await homePage.goTo(page, global.FO.URL);
+    const result = await homePage.isHomePage(page);
+    await expect(result).to.be.true;
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/06_homePage/01_checkSlider.js]:
```suggestion
    await homePage.goTo(page, global.FO.URL);
    
    const result = await homePage.isHomePage(page);
    await expect(result).to.be.true;
```

File: tests/UI/pages/FO/home/index.js [hunk 2]
```diff
@@ -83,6 +88,40 @@ class Home extends FOBasePage {
...
+  /**
+   * Is slider visible
+   * @param page {Page} Browser tab
+   * @param position {number} The slider position
+   * @returns {Promise<boolean>}
+   */
+  async isSliderVisible(page, position) {
+    await this.waitForVisibleSelector(page, this.carouselSliderId);
+
+    return this.elementVisible(page, `${this.carouselSliderInner}:nth-child(${position})`, 1000);
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/home/index.js]: Maybe use selectors for this (see other comment)

File: tests/UI/pages/FO/home/index.js [hunk 3]
```diff
@@ -83,6 +88,40 @@ class Home extends FOBasePage {
...
+  }
+
+  /**
+   * Clich in slider number
+   * @param page {Page} Browser tab
+   * @param position {number} The slider position
+   * @returns {Promise<string>}
+   */
+  async clickInSlider(page, position) {
+    await this.clickAndWaitForNavigation(page, `${this.carouselSliderInner}:nth-child(${position})`);
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/home/index.js]: same here

File: tests/UI/pages/FO/home/index.js [hunk 4]
```diff
@@ -16,6 +16,11 @@ class Home extends FOBasePage {
 
     this.pageTitle = global.INSTALL.SHOP_NAME;
 
+    // Selectors of slider
+    this.carouselSliderId = '#carousel';
+    this.carouselControlDirection = direction => `${this.carouselSliderId} a.${direction}.carousel-control`;
+    this.carouselSliderInner = `${this.carouselSliderId} ul.carousel-inner li`;
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/home/index.js]: Maybe 2 points Here : 

1. Name better your selectors
2. Add selector for `nth-child`

```suggestion
    this.carouselControlDirectionLink = direction => `${this.carouselSliderId} a.${direction}.carousel-control`;
    this.carouselSliderInnerList = `${this.carouselSliderId} ul.carousel-inner`;
    this.carouselSliderInnerListItems = `${this.carouselSliderInnerList} li`;
    this.carouselSliderInnerListItem = position => `${this.carouselSliderInnerListItems}:nth-child(${position})`;
```

File: tests/UI/campaigns/functional/FO/06_homePage/01_checkSlider.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    await expect(isVisible).to.be.true;
+  });
+
+  it('should click in the slider and check the URL', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'clickOnSlider', baseContext);
+
+    const currentURL = await homePage.clickInSlider(page, 2);
+    await expect(currentURL)
+      .to.contains('https://www.prestashop.com/en')
+      .and.to.contains('homeslider&utm_campaign=back-office-EN&utm_content=download');
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/FO/06_homePage/01_checkSlider.js:70]: Maybe add vars for this (URL could change)

**Review by Progi1984 — COMMENTED** [review]
A small change

File: tests/UI/campaigns/functional/FO/06_homePage/01_checkSlider.js (hunk 5 above)
Comment (Progi1984) [reply tests/UI/campaigns/functional/FO/06_homePage/01_checkSlider.js:70]: (I create an issue for the split :  https://github.com/PrestaShop/PrestaShop/issues/29836)

Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/FO/06_homePage/01_checkSlider.js:70]: Thank you @Progi1984 :100:

File: tests/UI/campaigns/functional/FO/06_homePage/01_checkSlider.js (hunk 1 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/FO/06_homePage/01_checkSlider.js]: Done thx :+1:

File: tests/UI/pages/FO/home/index.js (hunk 2 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/FO/home/index.js]: Done thx :+1:

**Review by Progi1984 — DISMISSED** [review]

**Review by boubkerbribri — DISMISSED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Thank you @nesrineabdmouleh

**Pull request merged**